File: CStoreManagement/app/src/main/java/com/c_store/zhiyazhang/cstoremanagement/view/activity/HomeActivity.java
package com.c_store.zhiyazhang.cstoremanagement.view.activity;

import android.annotation.SuppressLint;
import android.app.ActivityOptions;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AlertDialog;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.Toast;

import com.c_store.zhiyazhang.cstoremanagement.R;
import com.c_store.zhiyazhang.cstoremanagement.utils.activity.MyActivity;

import butterknife.BindView;
import butterknife.OnClick;

import static com.c_store.zhiyazhang.cstoremanagement.utils.static_variable.MyVariable.EXIT_APP_ACTION;

/**
 * Created by zhiya.zhang
 * on 2017/6/6 9:42.
 */

@SuppressLint("Registered")
public class HomeActivity extends MyActivity {

    @BindView(R.id.my_toolbar)
    Toolbar myToolbar;
    @BindView(R.id.gg1)
    LinearLayout gg1;
    @BindView(R.id.gg2)
    LinearLayout gg2;
    @BindView(R.id.gg3)
    LinearLayout gg3;
    @BindView(R.id.gg4)
    LinearLayout gg4;
    @BindView(R.id.gg5)
    LinearLayout gg5;
    @BindView(R.id.gg6)
    LinearLayout gg6;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initView();
    }

    private void initView() {
        myToolbar.setTitle(getResources().getString(R.string.app_name));
        myToolbar.setLogo(R.mipmap.ic_logo);
        setSupportActionBar(myToolbar);
    }

    @OnClick({R.id.gg1, R.id.gg2, R.id.gg3, R.id.gg4, R.id.gg5, R.id.gg6})
    public void Onclick(View view) {
        switch (view.getId()) {
            case R.id.gg1:
                Toast.makeText(this, "未完成", Toast.LENGTH_SHORT).show();
                break;
            case R.id.gg2:
                Toast.makeText(this, "未完成", Toast.LENGTH_SHORT).show();
                break;
            case R.id.gg3:
                startActivity(new Intent(HomeActivity.this, ContractTypeActivity.class),
                        ActivityOptions.makeSceneTransitionAnimation(HomeActivity.this, gg3, "gg3").toBundle());
                break;
            case R.id.gg4:
                Toast.makeText(this, "未完成", Toast.LENGTH_SHORT).show();
                break;
            case R.id.gg5:
                Toast.makeText(this, "未完成", Toast.LENGTH_SHORT).show();
                break;
            case R.id.gg6:
                Toast.makeText(this, "未完成", Toast.LENGTH_SHORT).show();
                break;
            default:
                break;
        }
    }

    /*
* 截取返回键做退出
* */
    @Override
    public void onBackPressed() {
        new AlertDialog.Builder(this)
                .setTitle("提示")
                .setMessage("请确认退出系统？")
                .setPositiveButton("退出", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        Intent intent = new Intent();
                        intent.setAction(EXIT_APP_ACTION);
                        sendBroadcast(intent);//发送退出系统广播  每个接收器都会收到 调动finish（）关闭activity
                        finish();
                    }
                })
                .setNegativeButton("按错了", null)
                .show();
    }


    @Override
    protected int getLayoutId() {
        return R.layout.activity_home2;
    }
}

File: KotlinCStoreManagement/app/src/main/java/com/cstore/zhiyazhang/cstoremanagement/view/HomeActivity.kt
package com.cstore.zhiyazhang.cstoremanagement.view

import android.annotation.SuppressLint
import android.app.ActivityOptions
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import android.support.design.widget.NavigationView
import android.support.v4.view.GravityCompat
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AlertDialog
import android.view.MenuItem
import android.widget.Toast
import com.cstore.zhiyazhang.cstoremanagement.R
import com.cstore.zhiyazhang.cstoremanagement.utils.MyActivity
import com.cstore.zhiyazhang.cstoremanagement.utils.MyToast
import com.cstore.zhiyazhang.cstoremanagement.view.contract.ContractOrder
import com.zhiyazhang.mykotlinapplication.utils.MyApplication
import kotlinx.android.synthetic.main.activity_home.*
import kotlinx.android.synthetic.main.app_bar_home.*
import kotlinx.android.synthetic.main.content_home.*
import kotlinx.android.synthetic.main.nav_header_home.view.*

class HomeActivity(override val layoutId: Int = R.layout.activity_home) : MyActivity(), NavigationView.OnNavigationItemSelectedListener {
    var updateButton = false//确认是否是通过更新按钮更新的
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initView()
        initListener()
        initReceiver()
    }

    private val updateReceiver: BroadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            if (intent.getBooleanExtra("is_new", false)) MyToast.getShortToast("发现新版本，正在下载中") else {
                if (updateButton) {
                    MyToast.getShortToast("已经是最新版本")
                    updateButton = false
                }
            }
        }
    }

    private fun initReceiver() {
        val intentFilter = IntentFilter()
        intentFilter.addAction("com.cstore.zhiyazhang.UPDATE")
        registerReceiver(updateReceiver, intentFilter)
    }

    override fun onStart() {
        super.onStart()
        this.startService(Intent(this, UpdateService::class.java))
    }

    override fun onDestroy() {
        unregisterReceiver(updateReceiver)
        super.onDestroy()
    }

    @SuppressLint("SetTextI18n")
    private fun initView() {
        toolbar.title = resources.getString(R.string.app_name)
        setSupportActionBar(toolbar)
        val toggle = ActionBarDrawerToggle(
                this, drawer_layout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close)
        drawer_layout.addDrawerListener(toggle)
        toggle.syncState()
        nav_view.setNavigationItemSelectedListener(this)
        val userShared = getSharedPreferences("user", Context.MODE_PRIVATE)
        val headerLayout = nav_view.inflateHeaderView(R.layout.nav_header_home)
        headerLayout.user_name.text = userShared.getString("uName", "")
        headerLayout.store_name.text = userShared.getString("storeName", "")
        headerLayout.store_address.text = userShared.getString("address", "")
        headerLayout.app_version.text = "v: ${MyApplication.getVersion()}"
    }

    private fun initListener() {
        gg1.setOnClickListener { MyToast.getShortToast("未完成") }
        gg2.setOnClickListener { MyToast.getShortToast("未完成") }
        gg3.setOnClickListener {
            startActivity(Intent(this@HomeActivity, ContractOrder::class.java),
                    ActivityOptions.makeSceneTransitionAnimation(this@HomeActivity, gg3, "gg3").toBundle())
        }
        gg4.setOnClickListener { MyToast.getShortToast("未完成") }
        gg5.setOnClickListener { MyToast.getShortToast("未完成") }
        gg6.setOnClickListener { MyToast.getShortToast("未完成") }
    }

    override fun onBackPressed() {
        AlertDialog.Builder(this)
                .setTitle("提示")
                .setMessage("请确认退出系统？")
                .setPositiveButton("退出") { _, _ ->
                    val intent = Intent()
                    intent.action = EXIT_APP_ACTION
                    sendBroadcast(intent)//发送退出系统广播  每个接收器都会收到 调动finish（）关闭activity
                    if (drawer_layout.isDrawerOpen(GravityCompat.START)) {
                        drawer_layout.closeDrawer(GravityCompat.START)
                    }
                    finish()
                }
                .setNegativeButton("按错了", null)
                .show()
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.nav_manage -> {
                Toast.makeText(this@HomeActivity, "开发中", Toast.LENGTH_SHORT).show()
            }
            R.id.nav_about -> {
                Toast.makeText(this@HomeActivity, "当前版本号：" + MyApplication.getVersion(), Toast.LENGTH_SHORT).show()
            }
            R.id.nav_update -> {
                updateButton = true
                this@HomeActivity.startService(Intent(this@HomeActivity, UpdateService::class.java))
            }
        }
        drawer_layout.closeDrawer(GravityCompat.START)
        return true
    }
}
